API.yaml

```diff
@@ -0,0 +1,949 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ActionRuleDTO'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: The action plan or the action type weren't found.
+      x-codegen-request-body-name: actionRulePostRequestDTO
+  '/actionrules/actionplan/{actionplanid}':
+    get:
+      summary: Gets action rules for an action plan.
+      description: Returns all action rules related to the specified action plan.
+      tags:
+        - action-rules
+      operationId: findActionRulesByActionPlanIdUsingGET
+      parameters:
+        - name: actionplanid
+          in: path
+          description: actionplanid
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Zero or more action rules were found for the action plan.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ActionRuleDTO'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: An action plan with the specified ID wasn't found.
+  '/actionrules/{actionRuleId}':
+    put:
+      summary: Updates an action rule.
+      description: Updates an action rule's priority, name, description and trigger time.
+      tags:
+        - action-rules
+      operationId: updateActionRuleUsingPUT
+      parameters:
+        - name: actionRuleId
+          in: path
+          description: actionRuleId
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: actionRulePutRequestDTO
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ActionRulePutRequestDTO'
+        required: false
+      responses:
+        '200':
+          description: The action rule was successfully updated.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ActionRuleDTO'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: An action rule with the specified ID wasn't found.
+        '500':
+          description: The ID provided wasn't a valid UUID.
+      x-codegen-request-body-name: actionRulePutRequestDTO
+    delete:
+      summary: Deletes an action rule.
+      description: Deletes the action rule by the specified ID.
+      tags:
+        - action-rules
+      operationId: deleteActionRuleUsingDELETE
+      parameters:
+        - name: actionRuleId
+          in: path
+          description: actionRuleId
+          required: true
+          schema:
+            type: string
+      requestBody:
```

Comment:
I thought you might have added this in error but then i checked the code.
wake me when its over.

Reply:
Yeah I mean, it sure violates principles to the degree that OpenAPI will moan about it

Reply:
Yeah I mean, it sure violates principles. So badly in fact that OpenAPI will moan about it